inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -654,12 +653,26 @@ void RemovePermutationsNHWCToNCHWPass::run() {
         auto next = getInputTo(l->outData.front()).begin()->second;
         auto prev = CNNNetPrevLayer(l);
 
-        if (!LayerInfo(next).isPermute() || !LayerInfo(prev).isPermute()) {
+        // The next layer must be NCHW to NHWC permute
+        if (!LayerInfo(next).isPermute() || next->GetParamAsInts("order") != std::vector<int>({0, 2, 3, 1})) {
```

Comment:
Wouldn't it be better to check layout instead of order of channels?
Otherwise, the order of channels should be returned by utility function and you should check both input and output layout for permute